typescript/src/entity.ts

```diff
@@ -149,8 +205,147 @@ export class Entity {
       Promise<ListResponse<InstanceType<T>>> {
     const resource = `/${this.resourceName}/`;
     const fetchOptions: RequestOptions = {};
-    if (options && options.embed) {
-      fetchOptions.query = [['embed', JSON.stringify(options.embed)]];
+    fetchOptions.query = [];
+    if (options) {
+      if (options.embed) {
+        fetchOptions.query.push(['embed', JSON.stringify(options.embed)]);
+      }
+      if (options.offset !== undefined) {
+        fetchOptions.query.push(['offset', options.offset.toString()]);
+      }
+      if (options.limit !== undefined) {
+        fetchOptions.query.push(['limit', options.limit.toString()]);
+      }
+      if (options.q !== undefined) {
+        fetchOptions.query.push(['q', options.q]);
+      }
+      if (options.sort !== undefined) {
+        fetchOptions.query.push(['sort', options.sort]);
+      }
+      if (options.order !== undefined) {
+        if (options.order === SortOrder.ASCENDING) {
+          fetchOptions.query.push(['order', 'asc']);
+        } else {
+          fetchOptions.query.push(['order', 'desc']);
+        }
+      }
+      if (options.tab !== undefined) {
+        fetchOptions.query.push(['tab', options.tab]);
+      }
+      if (options.as !== undefined) {
+        fetchOptions.query.push(['as', options.as]);
+      }
+      if (options.state !== undefined) {
+        fetchOptions.query.push(['state', options.state]);
+      }
+      if (options.categoryId !== undefined) {
+        fetchOptions.query.push(['category_id', options.categoryId.toString()]);
+      }
+      if (options.inDomain !== undefined) {
+        fetchOptions.query.push(['in_domain', options.inDomain.toString()]);
+      }
+      if (options.inDomainRoles !== undefined) {
+        fetchOptions.query.push(['in_domain_roles',
+          options.inDomainRoles.join(',')]);
+      } 
+      if (options.asRole !== undefined) {
+        fetchOptions.query.push(['as_role', options.asRole.toString()]);
+      } 
+      if (options.publicOnly !== undefined) {
+        fetchOptions.query.push(['public_only', options.publicOnly.toString()]);
+      } 
+      if (options.managedOnly !== undefined) {
+        fetchOptions.query.push(['managed_only',
+          options.managedOnly.toString()]);
+      } 
+      if (options.memberOnly !== undefined) {
+        fetchOptions.query.push(['member_only', options.memberOnly.toString()]);
+      } 
+      if (options.domainRoles) {
```

Comment:
domainRoles does not need "options.domainRoles !== undefined"

Reply:
just want to mention in JS Boolean([]) equal to true, I do not think typescript changed that behaivor?

Reply:
the `[]` doesn't cause an observable bug, but you are correct, it should be more consisitent with the others

Reply:
`embed` and `withRights` are intentionally different